SubZero/src/util/filter/HSVFilter.h

```diff
@@ -0,0 +1,112 @@
+/**
+  * HSVFilter.h
+  * Created on: June 19, 2015
+  * Author: James534
+  */
+
+#ifndef HSVFILTER_H
+#define HSVFILTER_H
+
+#include "Filter.h"
+#include <opencv2/imgproc.hpp>
+
+/**
+ * The HSV Filter allows filtering of Hue, Saturation, Value
+ * Uses Opencv's built-in HSV filter, along with noise filtering functions
+ */
+class HSVFilter : public Filter{
+public:
+
+    /* ==================================================
+     * Constructors
+     * ==================================================
+     */
+    /**
+     * Empty HSV filter. Set up the filter values by the set___ functions
+     */
+    HSVFilter();
+
+    /**
+     * HSVFilter with all the parameters specified
+     * @param lowH  = lower bound of Hue
+     * @param highH = upper bound of Hue
+     * @param lowS  = lower bound of Saturation
+     * @param highS = upper bound of Saturation
+     * @param lowV  = lower bound of Value
+     * @param highV = upper bound of Value
+     */
+    HSVFilter(int lowH, int highH, int lowS, int highS, int lowV, int highV);
```

Comment:
in the RGBFilter I use array to group these values... should we be consistent with how we do this?
this is how I did mine: RGBFilter(array[3]) where array[3] = { r,g,b}... then when I need to separate to highs and lows I do RGB(array[3],array[3],array[3]) (the third one is for the mids)
I feel this is cleaner.. since for the latter constructor the user would have to input 9 separate args... though they are all ints... what do you think?